### PR TITLE
Add security helper for Security/Mvc usage

### PR DESCRIPTION
**Pull request by HaoK**
Needed to remove dependency on Auth from Mvc

cc @davidfowl @Tratcher

### CODE REVIEW

File: src/Microsoft.Framework.SecurityHelper.Sources/SecurityHelper.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+using System.Linq;
+using System.Security.Claims;
+
+namespace Microsoft.Framework.Internal
+{
+    /// <summary>
+    /// Helper code used when implementing authentication middleware
+    /// </summary>
+    public static class SecurityHelper
```
Comment (Tratcher) [inline src/Microsoft.Framework.SecurityHelper.Sources/SecurityHelper.cs]: The class should be internal for shared sources.

File: src/Microsoft.Framework.SecurityHelper.Sources/project.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    "version": "1.0.0-*",
+    "shared": "*.cs",
+    "dependencies": {
+    },
+    "frameworks": {
+        "net45": { },
+        "dnx451": { },
+        "dnxcore50": {
+            "dependencies": {
+                "System.Runtime": "4.0.20-*"
```
Comment (Tratcher) [inline src/Microsoft.Framework.SecurityHelper.Sources/project.json]: Shouldn't there be a dependency for Claims?

Comment (HaoK) [inline src/Microsoft.Framework.SecurityHelper.Sources/project.json]: It builds without it, so I didn't add it, but I had the same thought...

Comment (HaoK) [inline src/Microsoft.Framework.SecurityHelper.Sources/project.json]: We don't have a claims dependency in Security today either, but maybe someone else brings it in?

Comment (Tratcher) [inline src/Microsoft.Framework.SecurityHelper.Sources/project.json]: Http.Abstractions

Comment (HaoK) [inline src/Microsoft.Framework.SecurityHelper.Sources/project.json]: Ok so do we want to add it explicitly here in the security helper as well?

Comment (Tratcher) [inline src/Microsoft.Framework.SecurityHelper.Sources/project.json]: If it compiles already then just leave it.

File: src/Microsoft.Framework.SecurityHelper.Sources/SecurityHelper.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace Microsoft.Framework.Internal
+{
+    /// <summary>
+    /// Helper code used when implementing authentication middleware
+    /// </summary>
+    internal static class SecurityHelper
+    {
+        /// <summary>
+        /// Add all ClaimsIdenities from an additional ClaimPrincipal to the ClaimsPrincipal
```
Comment (nil4) [inline src/Microsoft.Framework.SecurityHelper.Sources/SecurityHelper.cs]: Typo: ClaimsIdenities -> ClaimsIdentities

File: src/Microsoft.Framework.SecurityHelper.Sources/SecurityHelper.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        public static ClaimsPrincipal MergeUserPrincipal([NotNull] ClaimsPrincipal existingPrincipal, [NotNull] ClaimsPrincipal additionalPrincipal)
+        {
+            var newPrincipal = new ClaimsPrincipal();
+            // New principal identities go first
+            newPrincipal.AddIdentities(additionalPrincipal.Identities);
+
+            // Then add any existing non empty or authenticated identities
+            if (existingPrincipal != null)
+            {
+                newPrincipal.AddIdentities(existingPrincipal.Identities.Where(i => i.IsAuthenticated || i.Claims.Count() > 0));
```
Comment (nil4) [inline src/Microsoft.Framework.SecurityHelper.Sources/SecurityHelper.cs]: `i.Claims.Any()`?

File: src/Microsoft.Framework.SecurityHelper.Sources/project.json (hunk 2 above)
Comment (rynowak) [inline src/Microsoft.Framework.SecurityHelper.Sources/project.json]: Adding these references to a shared-source package doesn't really have any impact. We've been leaving them out on purpose.

File: src/Microsoft.Framework.SecurityHelper.Sources/SecurityHelper.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    /// </summary>
+    internal static class SecurityHelper
+    {
+        /// <summary>
+        /// Add all ClaimsIdenities from an additional ClaimPrincipal to the ClaimsPrincipal
+        /// Merges a new claims principal, placing all new identities first, and eliminating
+        /// any empty unauthenticated identities from context.User
+        /// </summary>
+        /// <param name="identity"></param>
+        public static ClaimsPrincipal MergeUserPrincipal([NotNull] ClaimsPrincipal existingPrincipal, [NotNull] ClaimsPrincipal additionalPrincipal)
```
Comment (rynowak) [inline src/Microsoft.Framework.SecurityHelper.Sources/SecurityHelper.cs]: long line

File: src/Microsoft.Framework.SecurityHelper.Sources/SecurityHelper.cs (hunk 3 above)
Comment (rynowak) [inline src/Microsoft.Framework.SecurityHelper.Sources/SecurityHelper.cs]: If you're going to take the trouble to write docs you can just use `<see cref="..." />`

**Comment by HaoK** [comment]
b9bfbd5a77d58fe0d54251908aa4a960c0913a71

**Pull request closed without merging**